bin/snap-sync

```diff
@@ -237,14 +243,25 @@ done
 i=0
 disk=-1
 disk_count=0
-for x in $UUIDS; do
-    UUIDS_ARRAY[$i]=$x
-    if [[ "$x" == "$uuid_cmdline" && ${SUBVOLIDS_ARRAY[$((i))]} == "$subvolid_cmdline" ]]; then
-        disk=$i
-        disk_count=$(($disk_count+1))
-    fi
-    i=$((i+1))
-done
+if [[ -z "$mount_cmdline" ]]; then
+    for x in $UUIDS; do
+        UUIDS_ARRAY[$i]=$x
+        if [[ "$x" == "$uuid_cmdline" && ${SUBVOLIDS_ARRAY[$((i))]} == "$subvolid_cmdline" ]]; then
+            disk=$i
+            disk_count=$(($disk_count+1))
+        fi
+        i=$((i+1))
+    done
+else
+    for x in $UUIDS; do
+        UUIDS_ARRAY[$i]=$x
+        if [[ "$x" == "$uuid_cmdline" && ${SUBVOLIDS_ARRAY[$((i))]} == "$subvolid_cmdline" && ${TARGETS_ARRAY[$((i))]} == "$mount_cmdline" ]]; then
```

Comment:
I would do something like:
```
if [[ -n "$mount_cmdline" && ${TARGETS_ARRAY[$((i))]} != "$mount_cmdline" ]]; then
    continue
fi
```

This should make the code smaller, but **I did not test it**.